src/pal/src/synchmgr/synchcontrollers.cpp

```diff
@@ -556,16 +556,16 @@ namespace CorUnix
     Sets the signal count of the target object, possibly triggering
     waiting threads awakening.
     --*/
-    PAL_ERROR CSynchStateController::SetSignalCount(DWORD dwNewCount)
+    PAL_ERROR CSynchStateController::SetSignalCount(LONG lNewCount)
     {
         VALIDATEOBJECT(m_psdSynchData);
 
         _ASSERTE(InternalGetCurrentThread() == m_pthrOwner);
-        _ASSERT_MSG((DWORD)INT_MAX > dwNewCount,
+        _ASSERT_MSG((ULONG)INT_MAX >= (ULONG)lNewCount,
```

Comment:
Since lNewCount is a signed integer now and INT_MAX is a valid value, should we just assert that it is >=0?